dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/clouddumper/Main.java

```diff
@@ -84,6 +118,22 @@ static class ExtractorConfiguration {
   static class ConnectorConfiguration {
 
     private String connector;
+
     private List<String> args;
   }
+
+  private static final String decrypt(Optional<CryptoKeyName> keyName, String input) {
+    if (!keyName.isPresent()) {
+      LOG.info("Got no decryption key. Using input as is.");
+      return input;
+    }
+    LOG.info("Using key {} to decrypt.", keyName);
+    ByteString ciphertext = ByteString.copyFrom(Base64.decodeBase64(input));
+    try (KeyManagementServiceClient client = KeyManagementServiceClient.create()) {
+      DecryptResponse response = client.decrypt(keyName.get(), ciphertext);
+      return response.getPlaintext().toString(StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
```

Comment:
I'd always rather let an exception propagate as a checked exception than catch it and mask it as a runtime.

Reply:
Understandable, but in this cases it is needed because `Stream.map()` takes a `Function` which cannot propagate checked exceptions.